sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,196 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+
+#include "sw/device/lib/base/mmio.h"
+
+// TODO - Should be autogenerated.
+// - Should padctrl/pinmux share some defines/constants (such as MIO_NUM) or do
+// we just autogen the same number twice?
+#define DIF_PADCTRL_DIO_NUM 32
+#define DIF_PADCTRL_MIO_NUM 32
+#define DIF_PADCTRL_ATTR_WIDTH 8
+#define DIF_PADCTRL_ADDITIONAL_ATTR_WIDTH 2
+
+/**
+ * Pad attributes
+ *
+ * Some attributes have defined meanings, others are implementation defined
+ * (|kDifPadCtrlAttrAdditionalFirst| and onwards). not all pads will support all
+ * attributes.
+ */
+typedef enum dif_padctrl_attr {
+  kDifPadctrlAttrIOInvert = 0,
+  kDifPadctrlAttrOpenDrain,
+  kDifPadctrlAttrPullDown,
+  kDifPadctrlAttrPullUp,
+  kDifPadctrlAttrKeeper,
+  kDifPadctrlAttrWeakDrive
+  kDifPadctrlAttrAdditionalFirst = 0,
```

Comment:
What are the first and last variants for? It's kind of hard to justify them without also including the corresponding .c file.